onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -30,31 +30,31 @@ class SVMCommon {
   KERNEL get_kernel_type() const { return kernel_type_; }
 
   float kernel_dot(const T* A, int64_t a, const std::vector<float>& B, int64_t b, int64_t len, KERNEL k) const {
-    float sum = 0.f;
+    double sum = 0.f;
+    const T* pA = A + a;
+    const float* pB = &B[b];
     if (k == KERNEL::POLY) {
-      for (int64_t i = 0; i < len; i++) {
-        sum += B[b + i] * static_cast<float>(A[a + i]);
-      }
+      for (int64_t i = len; i > 0; --i, ++pA, ++pB)
+        sum += *pB * static_cast<float>(*pA);
```

Comment:
>static_cast<float>(*pA); [](start = 21, length = 24)

The cast seems to be wrong everywhere. One can not cast to float in this function. T exists to represent a specific type not float.

Reply:
removed